src/webgpu/print_environment.spec.ts

```diff
@@ -36,8 +38,19 @@ in the logs. On non-WPT runtimes, it will also print to the console with console
 WPT disallows console.log and doesn't support logs on passing tests, so this does nothing on WPT.`
   )
   .fn(t => {
-    const isCompatibilityMode = (t.adapter as unknown as { isCompatibilityMode?: boolean })
-      .isCompatibilityMode;
+    // `t.device` will be the default device, because no additional capabilities were requested.
+    const defaultDeviceProperties = Object.fromEntries(
+      (function* () {
+        const device = t.device as unknown as Record<string, unknown>;
+        for (const key in device) {
+          // Skip things that we don't want to JSON.stringify.
+          if (['lost', 'queue', 'onuncapturederror', 'label'].includes(key)) {
+            continue;
+          }
+          yield [key, device[key]];
+        }
+      })()
+    );
```

Comment:
Looking at what is actually used eventually, how about simply filtering `features` and `limits` only? If we end up adding more stuff, we may to update this code to add new things to filter out, so we may want to be explicit now.

```suggestion
```

Reply:
That's how `adapter` was before, but I don't think there's any harm in printing all the members (even printing these filtered-out ones would be fine, we just know for sure that it's not useful). In practice I don't think it's going to be kept up to date otherwise. If we do it this way, then we won't have to wait for someone to notice that something is missing from this test, and then actually bother to fix it.

Reply:
Gotcha!